### PR TITLE
OCPBUGS-28538: Added docs to get other sdn providers deployed with agent provider

### PR DESCRIPTION
**Pull request by mvazquezc**
**What this PR does / why we need it**:
This PR adds docs to the agent provider section on how to get other SDN providers (Calico and Cilium) deployed in a HostedCluster.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
None

**Checklist**
- [X] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [X] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: docs/content/how-to/agent/other-sdn-providers.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
+# Create an Agent cluster with other SDN Providers
+
+This document explains how to create a HostedCluster that runs an SDN provider different from OVNKubernetes. The document assumes that you already have the required infrastructure in place to create HostedClusters using the Agent provider. If not, please refer to the [Agent guide](./create-agent-cluster.md) and come back later.
+
+> **IMPORTANT**: The work described here is **not supported**. SDN providers **must** certify their software on HyperShift before it becomes a supported solution. The steps described here are just a technical reference for people that wants to try different SDN providers in HyperShift.
```
Comment (bryan-cox) [inline docs/content/how-to/agent/other-sdn-providers.md:5]:
```suggestion
> **IMPORTANT**: The work described here is **not supported**. SDN providers **must** certify their software on HyperShift before it becomes a supported solution. The steps described here are just a technical reference for people who want to try different SDN providers in HyperShift.
```

File: docs/content/how-to/agent/other-sdn-providers.md
Comment (bryan-cox) [inline docs/content/how-to/agent/other-sdn-providers.md:1]: You'll need to modify mkdocs.yml as well to include this in the directory. https://github.com/openshift/hypershift/blob/54242ed74dc152e24612d68d77e0381fc7a9aea2/docs/mkdocs.yml#L78

**Comment by jparrill** [comment]
Handing this PR over meanwhile @mvazquezc is OoO.

**Comment by jparrill** [comment]
/title OCPBUGS-26977,OCPBUGS-28538: Added docs to get other sdn providers deployed with agent provider

**Comment by jparrill** [comment]
/retitle OCPBUGS-28538: Added docs to get other sdn providers deployed with agent provider

File: docs/content/how-to/sdn/other-sdn-providers.md [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+## Calico
+### Deployment
+
+In this scenario we are using the Calico version v3.27.0 which is the last one at the time of this writing. The steps followed rely on the [docs](https://docs.tigera.io/calico/latest/getting-started/kubernetes/openshift/installation#generate-the-install-manifests) by Tigera to deploy Calico on OpenShift.
+
+1. Create a `HostedCluster` and set its `HostedCluster.spec.networking.networkType` to `Calico`.
+
+2. Wait for the HostedCluster's API to be ready. Once it's ready, get the admin kubeconfig.
+
+3. Wait for compute nodes to join the cluster. Keep in mind since the SDN is not deployed yet, they will remain in `NotReady` state.
```
Comment (csrwng) [inline docs/content/how-to/sdn/other-sdn-providers.md:19]: I don't think it's necessary to wait for nodes to join. You should be able to apply the manifests as long as you have a working API server.

File: docs/content/how-to/sdn/other-sdn-providers.md [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+## Cilium
+### Deployment
+
+In this scenario we are using the Cilium version v1.14.5 which is the last one at the time of this writing. The steps followed rely on the [docs](https://docs.cilium.io/en/stable/installation/k8s-install-openshift-okd/) by Cilium project to deploy Cilium on OpenShift.
+
+1. Create a `HostedCluster` and set its `HostedCluster.spec.networking.networkType` to `Other`.
+
+2. Wait for the HostedCluster's API to be ready. Once it's ready, get the admin kubeconfig.
+
+3. Wait for compute nodes to join the cluster. Keep in mind since the SDN is not deployed yet, they will remain in `NotReady` state.
```
Comment (csrwng) [inline docs/content/how-to/sdn/other-sdn-providers.md:143]: Same as above, no need to wait for nodes.

**Review by csrwng — COMMENTED** [review]
a couple of comments

File: docs/content/how-to/sdn/other-sdn-providers.md (hunk 2 above)
Comment (jparrill) [reply docs/content/how-to/sdn/other-sdn-providers.md]: fixed

File: docs/content/how-to/sdn/other-sdn-providers.md (hunk 3 above)
Comment (jparrill) [reply docs/content/how-to/sdn/other-sdn-providers.md]: fixed

**Comment by jparrill** [comment]
The backport is not needed because it's only documentation, but the affectation it's also covering 4.14 OCP versions.

**Pull request merged**